drivers/Ticker.h

```diff
@@ -121,8 +125,21 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
      *  for threads scheduling.
      *
      */
+    MBED_DEPRECATED_SINCE("mbed-os-6.0.0", "Pass a chrono duration, not an integer microsecond count. For example use `10ms` rather than `10000`.")
     void attach_us(Callback<void()> func, us_timestamp_t t);
 
+    /** Attach a function to be called by the Ticker, specifying the interval in microseconds
+     *
+     *  @param func pointer to the function to be called
+     *  @param t the time between calls in micro-seconds
+     *
+     *  @note setting @a t to a value shorter than it takes to process the ticker callback
+     *  causes the system to hang. Ticker callback is called constantly with no time
+     *  for threads scheduling.
+     *
+     */
+    void attach(Callback<void()> func, TickerClock::duration t);
```

Comment:
This is possibly the most important design question. It applies both to `TickerClock::duration` and `Kernel::Clock::duration[_u32]`.  

Writing it like this potentially leaves open the door for changing what the API frequency is. I can envisage doing that. For the ticker, it could be done to matching hardware frequency, which would reduce run-time conversions; for the OS it might be part of changing the tick speed for power reasons, and then minimising conversions at run-time again.

Actually doing such a change would not be smooth though - for example if the tick rate was lowered to 100Hz, then you can't just change the `period`, as 

    ThisThread::sleep_for(100ms);

would fail to compile, as you can't implicitly assign `milliseconds` to centiseconds, even if an exact constant like that.

You would need to probably need to change the APIs to be template form to handle rounding if you did that.

We did previously rule that the RTOS tick rate was fixed at milliseconds, and documented that. This is something of a step back from that.

And it is just to provide scope to minimise run-time conversions. We could alternatively lock these APIs and clocks as Chrono milliseconds and microseconds, with commitment to never change (regardless of actual RTOS tick rate or hardware rate), and accept run-time conversion cost.

Oh, there is also a separate symmetry issue, in that time points must be expressed as `TickerClock::time_point`. So whether or not `TickerClock::duration` is fixed, writing it as `TickerClock::duration` just looks neat next to the `time_point`, and makes explicit that this matches the value you'd get by, say, subtracting two `TickerClock::now` values. No conversion needed.

Reply:
That makes sense, thanks.